sympy/series/order.py

```diff
@@ -385,6 +385,8 @@ def contains(self, expr):
             for s in common_symbols:
                 l = ratio.limit(s, point)
                 from sympy.series.limits import Limit
+
+                l = Limit(ratio, s, point).doit(heuristics=False)
```

Comment:
What is a test that can be added that will differ in result if this line is not here?

Reply:
At first sight, it doesn't look like there's a difference. Maybe @skirpichev had some wider plans in mind. Anyway, _diofant_ appears to be much faster than _SymPy_ at calculating limits, so I think it's worth porting @skirpichev 's improvements back to _SymPy_.

Reply:
> At first sight, it doesn't look like there's a difference.

There is.  Now O.contains doesn't use heuristics, but algorithm.

> diofant appears to be much faster than SymPy at calculating limits

If you just worry about speed, I bet - sympy dev's could add a lot of new hacks which make your current code faster than diofant's.